### PR TITLE
Improve inventory task dumper at pipeline

### PR DESCRIPTION
**Pull request by azexcy**
Improve inventory task dumper to support more scenarios

Changes proposed in this pull request:
  - Improve inventory task dumper at pipeline

---

Before committing this PR, I'm sure that I have checked the following options:
- [ ] My code follows the [code of conduct](https://shardingsphere.apache.org/community/en/involved/conduct/code/) of this project.
- [ ] I have self-reviewed the commit code.
- [ ] I have (or in comment I request) added corresponding labels for the pull request.
- [ ] I have passed maven check locally : `./mvnw clean install -B -T1C -Dmaven.javadoc.skip -Dmaven.jacoco.skip -e`.
- [ ] I have made corresponding changes to the documentation.
- [ ] I have added corresponding unit tests for my changes.

### CODE REVIEW

File: kernel/data-pipeline/core/src/main/java/org/apache/shardingsphere/data/pipeline/core/ingest/dumper/InventoryDumper.java [hunk 1]
```diff
@@ -188,9 +190,12 @@ private DataRecord loadDataRecord(final ResultSet resultSet, final ResultSetMeta
         DataRecord result = new DataRecord(newPosition(resultSet), columnCount);
         result.setType(IngestDataChangeType.INSERT);
         result.setTableName(dumperConfig.getLogicTableName());
+        List<String> insertColumnNames = Optional.ofNullable(dumperConfig.getInsertColumnNames()).orElse(Collections.emptyList());
```
Comment (sandynz) [inline kernel/data-pipeline/core/src/main/java/org/apache/shardingsphere/data/pipeline/core/ingest/dumper/InventoryDumper.java:193]: Could we verify `resultSetMetaData.getColumnCount()` is equals with `dumperConfig.getInsertColumnNames().size`?

File: kernel/data-pipeline/core/src/main/java/org/apache/shardingsphere/data/pipeline/core/task/InventoryIncrementalTasksRunner.java [hunk 2]
```diff
@@ -49,6 +49,7 @@ public class InventoryIncrementalTasksRunner implements PipelineTasksRunner {
     
     private final Collection<IncrementalTask> incrementalTasks;
     
+    @Getter
     private final PipelineJobAPI jobAPI;
```
Comment (sandynz) [inline kernel/data-pipeline/core/src/main/java/org/apache/shardingsphere/data/pipeline/core/task/InventoryIncrementalTasksRunner.java:53]: Is `@Getter` required?

**Review by sandynz — CHANGES_REQUESTED** [review]

File: kernel/data-pipeline/core/src/main/java/org/apache/shardingsphere/data/pipeline/core/task/InventoryIncrementalTasksRunner.java (hunk 2 above)
Comment (azexcy) [reply kernel/data-pipeline/core/src/main/java/org/apache/shardingsphere/data/pipeline/core/task/InventoryIncrementalTasksRunner.java:53]: Maybe subclass only have inventory task, then it's will use jobAPI methods, just like `stop()`

File: kernel/data-pipeline/core/src/main/java/org/apache/shardingsphere/data/pipeline/core/ingest/dumper/InventoryDumper.java (hunk 1 above)
Comment (azexcy) [reply kernel/data-pipeline/core/src/main/java/org/apache/shardingsphere/data/pipeline/core/ingest/dumper/InventoryDumper.java:193]: image
The `resultSetMetaData.getColumnCount()` depends on querySQL, if provider querySQL, must specify `dumperConfig.getInsertColumnNames()`,  is there a better way to handle it?

Comment (sandynz) [reply kernel/data-pipeline/core/src/main/java/org/apache/shardingsphere/data/pipeline/core/ingest/dumper/InventoryDumper.java:193]: Just verify it when `dumperConfig.getInsertColumnNames()` is available

**Review by sandynz — APPROVED** [review]

**Pull request merged**